### PR TITLE
Improve Java logging

### PR DESCRIPTION
**Pull request by raulchen**
## Why are these changes needed?

- Config Java logging via RayConfig programmatically, so users won't be aware of log4j.
- Remove unnecessary log file suffixes.
- Reduce verbosity of some logs. 

## Related issue number

#7291 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: java/runtime/src/main/java/io/ray/runtime/DefaultRayRuntimeFactory.java [hunk 1]
```diff
@@ -27,7 +38,7 @@ public RayRuntime createRayRuntime() {
       runtime.start();
       return runtime;
     } catch (Exception e) {
-      LOGGER.error("Failed to initialize ray runtime", e);
+      logger.error("Failed to initialize ray runtime", e);
```
Comment (chaokunyang) [inline java/runtime/src/main/java/io/ray/runtime/DefaultRayRuntimeFactory.java:41]:
```suggestion
      logger.error("Failed to initialize ray runtime with config {}", rayConfig, e);
```

File: java/runtime/src/main/java/io/ray/runtime/runner/worker/DefaultDriver.java [hunk 2]
```diff
@@ -1,30 +0,0 @@
-package io.ray.runtime.runner.worker;
-
-import io.ray.api.Ray;
-
-/**
- * The main function of DefaultDriver.
- */
-public class DefaultDriver {
```
Comment (kfstorm) [inline java/runtime/src/main/java/io/ray/runtime/runner/worker/DefaultDriver.java:8]: Why delete `DefaultDriver`?

**Review by kfstorm — COMMENTED** [review]
Nice PR!

File: java/runtime/src/main/java/io/ray/runtime/runner/worker/DefaultDriver.java (hunk 2 above)
Comment (raulchen) [reply java/runtime/src/main/java/io/ray/runtime/runner/worker/DefaultDriver.java:8]: It's not actually used. The the code is out-dated. I think we can re-add it when we introduce job submission.

**Comment by raulchen** [comment]
@kfstorm @chaokunyang thanks, I've addressed your comments.

**Review by kfstorm — APPROVED** [review]

**Pull request merged**